### DISCUSSION TITLE
Better rich text editor

### DISCUSSION THREAD

**Discussion by demshy** [decaporg/decap-cms#6905] [Ideas]
I am considering switching to Plate (https://platejs.org/) because as I see it, the plugin implementations are close enough to ours that it does not make sense to duplicate this stuff. 

Pros, from my POV:

- I’m pretty sure they already fixed a ton of bugs that we have not even encountered yet
- Their test coverage is more thorough
- This gives us a ton of additional plugins and features that we can implement in the future
- Because it is based on slate, we get to keep most of our serialisers (which Plate lacks) and most of the editor behaviour (but not all, see below)
- We get to keep shortcodes (yay)
- We can add tables plugin (double yay)

Things to consider / potential downsides:

- Some shortcuts (headings for example) are different by default and it might be painful to change them
- Our current implementation handles some edge cases on Enter or Backspace key differently: one example is within an empty heading which is the first element of the document - there are specific cypress rules for these two but I’m not sure how seriously attached the users are
- I’m not there yet, but I’m pretty sure a lot of specific list behaviour is different and cypress tests would need adaptation - same as the previous point

I’m already working on a proof of concept on the side and will be able to open a draft PR soon, but would appreciate any thoughts before digging deeper.

**Comment by iyansanjaya** [comment]
cool, if that possible and can improve what already exists, that would be great.

**Comment by TravisBrace** [comment]
This looks sweet! I'm currently looking at ways to add in a custom markdown widget. Just to add some of these rich text features that plate has. Like alignment. I really want to add more to the image insert component as well. being able to align or add a class to images would be huge. (full width, half width, float left, float right)

**Comment by TravisBrace** [comment]
If there's anyway I can help please let me know!!

**Reply by demshy** [reply]
I have been busy with some other matters lately, but I think I'll be able to continue in the following weeks. I'll get my current POC in shape so we have a decent base to work on, then I'll take all the help I can get :)

**Comment by naturedamends** [comment]
https://github.com/decaporg/decap-cms/discussions/6964
Mentions an issue with the rich text editor parsing on iPad safari.

**Comment by martinjagodic** [comment]
We should add math equations to the list of features that we want in the Slate replacement: https://github.com/decaporg/decap-cms/issues/1279

**Comment by naturedamends** [comment]
I know it may not be a RTE issues, but it must reside close. And it may have to incorporate usage of a text editor.

Are there plans for getting the diff, and conflict resolution?

**Comment by wadclapp** [comment]
Big +1 for this. It'd be a major improvement.

Any idea on when it might be implemented? I noticed there's a PR now for it here #7162 

I've just started using Decap, and some of the most important features it's missing for me are related to the editor:
- Missing: tables, text alignment, image insert/alignment, etc
- and the biggest one, missing the ability to add inline custom widgets (#5065),

I'm using Nuxt Content and I'd like to add inline components, and format text with styled spans and attributes ([docs](https://content.nuxt.com/usage/markdown#span-text)):
```md
Hello [World]{style="background-color: var(--color-primary-500)"}!

Hello [World]{style="color: green;" .custom-class #custom-id}!
```

Will this feature solve these problems?
